Doc 404 - doc client connectivity issue with NodePort Access on OpenShift (#1584)
hazelcast/hz-docs: docs/modules/kubernetes/pages/troubleshooting-and-limitations.adoc

----
helm install my-release ... --set jet.enabled=false hazelcast/<chart>
----

== Troubleshooting Kubernetes Auto Discovery

> === Client connectivity issue with NodePort access on OpenShift

Reviewer: Does this title and section need tweaking in response to Serge's feedback?

Author: tweaked

> Clients attempting to connect to a Hazelcast cluster deployed on OpenShift using the Helm chart and NodePort may face connectivity issues when accessing the cluster outside Kubernetes. Specifically, clients operating in smart/ALL_MEMBERS mode fetch internal pod IPs, which are inaccessible externally, can lead to connection failures after the initial handshake.

Reviewer: ```suggestion
Clients attempting to connect to a Hazelcast cluster deployed on OpenShift using the Helm chart and NodePort may face connectivity issues when accessing the cluster outside Kubernetes. Specifically, clients operating in smart/ALL_MEMBERS mode fetch internal pod IPs, which are inaccessible externally, leading to connection failures after the initial handshake.
```

> This issue arises due to Hazelcast's default behavior of using Kubernetes DNS mode when Role-Based Access Control (RBAC) permissions are not configured. This leads to internal pod IP discovery instead of external NodePort addresses. 

Reviewer: ```suggestion
This issue arises due to Hazelcast's default behavior of using Kubernetes DNS mode when Role-Based Access Control (RBAC) permissions are not configured.
```
I think you've covered this in the previous paragraph

> You can resolve this problem by enabling the required RBAC permissions. This allows the Hazelcast Kubernetes discovery plugin to retrieve external node addresses:

Reviewer: ```suggestion
You can resolve this problem by enabling the required RBAC permissions in the Helm chart. This allows the Hazelcast Kubernetes discovery plugin to retrieve external node addresses:
```
I'd be inclined to be even more specific (e.g. "add these lines to `Chart.yaml`") but this is probably clear enough and in line with the rest of the docs.

The user has presumably hit this problem after installing Hazelcast with the chart, so what do they need to do to apply this change? Recreate the cluster using the updated chart (how, exactly)? @SeriyBg
